ansible/roles/texlive/tasks/main.yml

```diff
@@ -0,0 +1,46 @@
+- name: texlive | prereq
+  yum: name={{ item }} state=latest
+  with_items:
+    - perl-Tk
+    - perl-Digest-MD5
+
+- name: texlive | create the temp install folder
+  file: path="{{ texlive_dest }}/tmp/" state=directory mode=0755
+    
+- name: texlive | download and extract installer
+  unarchive:
+    copy: no
+    # Checking whether this version of texlive is installed
+    # before downloading the installer.
+    # Can the 'unarchive' test for a directory?
+    creates: "{{ TEXDIR }}"
+    dest: "{{ texlive_dest }}/tmp/"
+    src: http://mirror.unl.edu/ctan/systems/texlive/tlnet/install-tl-unx.tar.gz
+
+- name: texlive | rename the installer folder
+  shell: mv install-tl-* install-tl-ansible
+  args:
+    chdir: "{{ texlive_dest }}/tmp/"
+    creates: "{{ texlive_dest }}/tmp/install-tl-ansible/"
```

Comment:
Again you'll want to skip this if texlive is already installed, so add a `creates`